runtime/src/main/java/com/dylibso/chicory/runtime/Instance.java

```diff
@@ -656,38 +655,33 @@ public Instance build() {
             var imports = new Import[importCount];
             for (int i = 0; i < importCount; i++) {
                 Import imprt = module.importSection().getImport(i);
-                switch (imprt.importType()) {
-                    case FUNCTION:
-                        {
-                            var type = ((FunctionImport) imprt).typeIndex();
-                            if (type >= this.module.typeSection().typeCount()) {
-                                throw new InvalidException("unknown type");
-                            }
-                            functionTypes[funcIdx] = type;
-                            // The global function id increases on this table
-                            // function ids are assigned on imports first
-                            imports[importId++] = imprt;
-                            funcIdx++;
-                            break;
-                        }
-                    default:
-                        imports[importId++] = imprt;
-                        break;
+                if (imprt.importType() == FUNCTION) {
+                    var type = ((FunctionImport) imprt).typeIndex();
+                    if (type >= this.module.typeSection().typeCount()) {
+                        throw new InvalidException("unknown type");
+                    }
+                    functionTypes[funcIdx] = type;
+                    // The global function id increases on this table
+                    // function ids are assigned on imports first
+                    imports[importId++] = imprt;
```

Comment:
This could be moved below and the else block removed

Reply:
right :+1: